src/test/commands.datasource.entry.test.ts

```diff
@@ -0,0 +1,743 @@
+			const newVal: string = '';
+			editDSEntryCommand.handleDataSourceEntryEdit(dvConfig, dsConfig, f, eName, newVal)
+				.then( (success) => {
+					if (success) {
+						dvConfig.spec.env.length.should.equal(oldLen);
+						should.exist(dvConfig.spec.env.find( (element) => {
+							return element.name === utils.generateFullDataSourceConfigEntryKey(dsConfig, eName) && element.value === newVal;
+						}));
+						done();
+					} else {
+						done(new Error('Execution of the Edit DataSource command returned false'));
+					}
+				})
+				.catch( (err) => {
+					done(err);
+				});
+		});
+
+		it('should return false when changing a datasource entry with an invalid model', (done) => {
+			const dvConfig: IDVConfig = utils.loadModelFromFile(f);
+			const dsConfig: IDataSourceConfig = {
+				name: dsName,
+				type: mongoDSType,
+				entries: new Map()
+			};
+			dvConfig.spec.env.forEach( (element: IEnv) => {
+				const key: string = element.name.slice(prefix.length+1);
+				dsConfig.entries.set(key, element.value);
+			});
+			const newVal: string = '';
+			editDSEntryCommand.handleDataSourceEntryEdit(undefined, dsConfig, f, eName, newVal)
+				.then( (success) => {
+					if (success) {
+						done(new Error('Execution of the Edit DataSource command returned true, but should not'));
+					} else {
+						done();
+					}
+				})
+				.catch( (err) => {
+					done(err);
+				});
+		});
+
+		it('should return false when changing a datasource entry with an invalid datasource config', (done) => {
+			const dvConfig: IDVConfig = utils.loadModelFromFile(f);
+			const dsConfig: IDataSourceConfig = {
+				name: dsName,
+				type: mongoDSType,
+				entries: new Map()
+			};
+			dvConfig.spec.env.forEach( (element: IEnv) => {
+				const key: string = element.name.slice(prefix.length+1);
+				dsConfig.entries.set(key, element.value);
+			});
+			const newVal: string = '';
+			editDSEntryCommand.handleDataSourceEntryEdit(dvConfig, undefined, f, eName, newVal)
+				.then( (success) => {
+					if (success) {
+						done(new Error('Execution of the Edit DataSource command returned true, but should not'));
+					} else {
+						done();
+					}
+				})
+				.catch( (err) => {
+					done(err);
+				});
+		});
+
+		it('should return false when changing a datasource entry with an file', (done) => {
+			const dvConfig: IDVConfig = utils.loadModelFromFile(f);
+			const dsConfig: IDataSourceConfig = {
+				name: dsName,
+				type: mongoDSType,
+				entries: new Map()
+			};
+			dvConfig.spec.env.forEach( (element: IEnv) => {
+				const key: string = element.name.slice(prefix.length+1);
+				dsConfig.entries.set(key, element.value);
+			});
+			const newVal: string = '';
+			editDSEntryCommand.handleDataSourceEntryEdit(dvConfig, dsConfig, undefined, eName, newVal)
+				.then( (success) => {
+					if (success) {
+						done(new Error('Execution of the Edit DataSource command returned true, but should not'));
+					} else {
+						done();
+					}
+				})
```

Comment:
Sonar is mentioning that this inlined function is repeated several times in the file. Would be better to factorize it.